### ISSUE TITLE
Serilog.Core.Sinks.RestrictedSink: System.NullReferenceException

### ISSUE THREAD

**Issue by kevinblake** [serilog-contrib/Serilog.Sinks.AmazonS3#8]
Using the following config:

```
            var levelSwitch = new LoggingLevelSwitch();
            levelSwitch.MinimumLevel = LogEventLevel.Information;

            Log.Logger = new LoggerConfiguration()
                .WriteTo.AmazonS3(
                    
                    "filename.txt",
                    "<removed-bucket-name>",
                    Amazon.RegionEndpoint.EUWest1,
                    "<removed>",
                    "<removed>",
                    restrictedToMinimumLevel: LogEventLevel.Verbose,
                    outputTemplate: "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level:u3}] {Message:lj}{NewLine}{Exception}",
                    fileSizeLimitBytes: 200,
                    levelSwitch: levelSwitch,
                    buffered: true,
                    rollingInterval: RollingInterval.Minute,
                    retainedFileCountLimit: 10,
                    autoUploadEvents: true
                )
                .CreateLogger();
```

I'm getting the following error:

> 2019-12-05T14:10:06.4965914Z Caught exception while emitting to sink Serilog.Core.Sinks.RestrictedSink: System.NullReferenceException: Object reference not set to an instance of an object.
>    at Serilog.Sinks.AmazonS3.RollingFileSink.Emit(LogEvent logEvent)
>    at Serilog.Sinks.AmazonS3.AmazonS3Sink.Emit(LogEvent logEvent)
>    at Serilog.Core.Sinks.RestrictedSink.Emit(LogEvent logEvent)
>    at Serilog.Core.Sinks.SafeAggregateSink.Emit(LogEvent logEvent)

Any idea what I'm doing wrong?

**Comment by SeppPenner** [comment]
It looks good from the first view. I will have to check this in detail.

Note to me: See https://github.com/serilog/serilog/blob/dev/src/Serilog/Core/Sinks/RestrictedSink.cs.

**Comment by SeppPenner** [comment]
I found the issue: `Action<Exception> failureCallback = null` --> An exception occurs and the failure callback is `null` per default.

**Comment by SeppPenner** [comment]
This is now fixed in version 1.0.5 (I hope). Feel free to re-open this issue if the fix doesn't work for you.
